client/flutter/lib/pages/onboarding/onboarding_page.dart

```diff
@@ -46,19 +69,52 @@ class _OnboardingPageState extends State<OnboardingPage> {
         physics: NeverScrollableScrollPhysics(),
         children: <Widget>[
           LegalLandingPage(onNext: _onLegalDone),
-          NotificationsPage(onNext: _toNextPage),
-          LocationSharingPage(onNext: _onDone),
+          CountrySelectPage(
+            onOpenCountryList: _toNextPage,
+            onNext: _onChooseCountry,
+            countryName: _selectedCountry?.name,
+          ),
+          if (_showCountryListPage)
+            CountryListPage(
+              countries: _countries,
+              onBack: _toPreviousPage,
+              selectedCountryCode: _selectedCountry?.alpha2Code,
+              onCountrySelected: _selectCountry,
+            ),
+          NotificationsPage(onNext: _onDone),
         ],
       ),
     );
   }
 
+  void _selectCountry(IsoCountry country) async {
+    _selectedCountry = country;
+    await setState(() {});
+    await _toPreviousPage();
```

Comment:
@davidkaneda I am automatically sending users back to the previous page as soon as they choose a country from the list. Also, the page transition looks like a "back" transition since technically we are navigating back a page. Let me know if those two behaviors are what you want here!

Reply:
Both are great, thanks for the heads up!